pkg/apis/v1alpha1/subnetport_types.go

```diff
@@ -0,0 +1,64 @@
+/* Copyright © 2021 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SubnetPortSpec defines the desired state of SubnetPort.
+type SubnetPortSpec struct {
+	// Subnet defines the parent Subnet name of the SubnetPort.
+	Subnet string `json:"subnet,omitempty"`
+	// SubnetSet defines the parent SubnetSet name of the SubnetPort.
+	SubnetSet string `json:"subnetSet,omitempty"`
+	// AttachmentRef refers to the virtual machine which the SubnetPort is attached.
+	AttachmentRef corev1.TypedLocalObjectReference `json:"attachmentRef,omitempty"`
```

Comment:
I checked corev1.TypedLocalObjectReference is defined as :
       APIGroup *string `json:"apiGroup,omitempty"`
	Kind     *string `json:"kind,omitempty"`
	Name     *string `json:"name,omitempty"`

Is there any difference between apiGroup and apiVersion?

Reply:
Changed it to `corev1.ObjectReference`